chain/network/src/routing/routing.rs

```diff
@@ -153,14 +158,12 @@ impl EdgeInner {
 
     /// Build the hash of the edge given its content.
     /// It is important that peer0 < peer1 at this point.
-    fn build_hash(peer0: &PeerId, peer1: &PeerId, nonce: u64) -> CryptoHash {
-        let mut buffer = Vec::<u8>::new();
-        let peer0: Vec<u8> = peer0.into();
-        buffer.extend_from_slice(peer0.as_slice());
-        let peer1: Vec<u8> = peer1.into();
-        buffer.extend_from_slice(peer1.as_slice());
-        buffer.write_u64::<LittleEndian>(nonce).unwrap();
-        hash(buffer.as_slice())
+    pub fn build_hash(peer0: &PeerId, peer1: &PeerId, nonce: u64) -> CryptoHash {
```

Comment:
"It is important that peer0 < peer1 at this point"

Not feeling good about making a function with caller-checked precondition `pub`. Can we keep it private? Or at least add an assertion. I guess, at least `debug_assert(peer0 < peer1)` would be helpful regardless.

Reply:
Ok, I added assert.